amqp: add optional TLS support

lucab opened the pull request:
This commit adds optional TLS support, using OpenSSL.
It is enabled by default and pull in an additional dependency on
the openssl crate.
However, if not needed this can be disabled with `cargo --no-default-features`.
It has been successfully tested against a third-party RabbitMQ (3.5.3) instance.

***

As I was not sure if the lack of TLS support, explicitly mentioned in the README, was a design decision, I implemented this part as an optional feature/dependency, enabled by default. This may also allow you to swap in a different TLS library later. Let me know if this is ok for you, or if you prefer to directly wire it.

Antti commented on src/session.rs:
```diff
@@ -80,12 +81,14 @@ impl Session {
         let mut url_parser = UrlParser::new();
         url_parser.scheme_type_mapper(scheme_type_mapper);
         let url = try!(url_parser.parse(url_string));
+        let tls = if url.scheme == "amqps" { true } else { false };
```
Do you not like `let tls = url.scheme == "amqps";` ?

lucab replied:
I find it too confusing/implicit, but I can settle on
`let tls = { url.scheme == "amqps" };`

Antti replied:
At least that one looks better than the former,
because to me it's like:
`if true { true } else { false}`

Antti commented:
The lack of TLS is not a design decision, it's just lack of time/was not needed at the moment.
This looks very good so far.

Antti commented:
We should document in the Readme, that you can disable TLS support with `cargo --no-default-features`

lucab commented:
> The lack of TLS is not a design decision, it's just lack of time/was not needed at the moment.

I see. Do you prefer to keep this as an optional feature or should I just remove all the `#cfg[tls]` stuff?

Antti commented:
Let's keep the cfg for now until we'll be certain on the TLS library. I prefer the native rust one instead of openssl bindings.

lucab commented:
Amended: added doc and changed boolean assignment style.

> I prefer the native rust one instead of openssl bindings.

Are you referring to a specific/existing one, or are you talking about a future / not-yet-there TLS library?

Antti commented:
I haven't tried existing ones yet, but they seem promising.
